### PR TITLE
Bug fix parcoords - selecting lines outside displayed range axis

### LINKED ISSUE #4086: Parcoords constraint range is stick to axis extremes when moving beyond range
Problem in moving a `parcoords` selection bar with `constraint-ranges` when the `constraintrange` value go beyond defined dimension `range`.
[codepen](https://codepen.io/MojtabaSamimi/pen/WVRWaj?editors=0010)

### PR DESCRIPTION
**Pull request by archmoj**
Fixes #4086.

@plotly/plotly_js 

[Codepen before](https://codepen.io/MojtabaSamimi/pen/WVRWaj?editors=0010)
[Codepen after](https://codepen.io/MojtabaSamimi/pen/rXyajY?editors=1000)

### CODE REVIEW

File: test/image/mocks/gl2d_parcoords_out-of-range_selected-above-below.json [hunk 1]
```diff
@@ -0,0 +1,821 @@
+{
+    "data": [
```
Comment (etpinard) [inline test/image/mocks/gl2d_parcoords_out-of-range_selected-above-below.json]: Which looks like:

image

currently

File: test/jasmine/tests/parcoords_test.js [hunk 2]
```diff
@@ -1496,8 +1496,7 @@ describe('parcoords constraint interactions', function() {
         .then(delay(noSnapDelay))
         .then(function() {
             expect(getDashArray(1)).toBeCloseToArray(newDashArray);
-            // TODO: ideally this would get clipped to [0, 9]...
```
Comment (etpinard) [inline test/jasmine/tests/parcoords_test.js]: @archmoj do you have any idea why someone wrote that it would have been "ideal" to have `[0, 9]` here?

Comment (archmoj) [reply test/jasmine/tests/parcoords_test.js]: Since the range was `computed from pixel space` the bottom range was not exactly zero i.e. `-0.1020`.

Comment (etpinard) [reply test/jasmine/tests/parcoords_test.js]: got it. Thanks for the info.

**Comment by etpinard** [comment]
As discussed in a private convo:

TODO:

- [ ] add one more jasmine test

**Comment by archmoj** [comment]
> As discussed in a private convo:
> 
> TODO:
> 
> * [ ]  add one more jasmine test

Done in https://github.com/plotly/plotly.js/pull/4087/commits/b30f07eb373ea07c388310b6292d60c77d97985d.

**Comment by etpinard** [comment]
Very nicely done :dancer: :dancer:

**Pull request merged**